tests/Aspire.Hosting.PostgreSQL.Tests/PostgresFunctionalTests.cs

```diff
@@ -0,0 +1,366 @@
+            });
+
+            hb.AddNpgsqlDataSource(db1.Resource.Name);
+
+            using (var host = hb.Build())
+            {
+                await host.StartAsync();
+
+                await pipeline.ExecuteAsync(
+                    async token =>
+                    {
+                        using var connection = host.Services.GetRequiredService<NpgsqlConnection>();
+                        await connection.OpenAsync(token);
+
+                        var command = connection.CreateCommand();
+                        command.CommandText = $"CREATE TABLE cars (brand VARCHAR(255)); INSERT INTO cars (brand) VALUES ('BatMobile'); SELECT * FROM cars;";
+                        var results = await command.ExecuteReaderAsync(token);
+
+                        Assert.True(results.HasRows);
+                    });
+            }
+
+            // Stops the container, or the Volume would still be in use
+            await app.StopAsync();
+        }
+
+        var builder2 = TestDistributedApplicationBuilder.Create();
+        usernameParameter = builder2.AddParameter("user");
+        passwordParameter = builder2.AddParameter("pwd");
+        builder2.Configuration["Parameters:user"] = username;
+        builder2.Configuration["Parameters:pwd"] = password;
+
+        var postgres2 = builder2.AddPostgres("pg", usernameParameter, passwordParameter);
+        var db2 = postgres2.AddDatabase(postgresDbName);
+        postgres2.WithDataVolume(volumeName);
+
+        using (var app = builder2.Build())
+        {
+            await app.StartAsync();
+
+            var hb = Host.CreateApplicationBuilder();
+
+            hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+            {
+                [$"ConnectionStrings:{db2.Resource.Name}"] = await db2.Resource.ConnectionStringExpression.GetValueAsync(default)
+            });
+
+            hb.AddNpgsqlDataSource(db2.Resource.Name);
+
+            using (var host = hb.Build())
+            {
+                await host.StartAsync();
+
+                await pipeline.ExecuteAsync(
+                    async token =>
+                    {
+                        using var connection = host.Services.GetRequiredService<NpgsqlConnection>();
+                        await connection.OpenAsync(token);
+
+                        var command = connection.CreateCommand();
+                        command.CommandText = $"SELECT * FROM cars;";
+                        var results = await command.ExecuteReaderAsync(token);
+
+                        Assert.True(results.HasRows);
+                    });
+            }
+
+            // Stops the container, or the Volume would still be in use
+            await app.StopAsync();
+        }
+
+        DockerUtils.AttemptDeleteDockerVolume(volumeName);
+    }
+
+    [Fact]
+    [RequiresDocker]
+    public async Task WithDataBindMountShouldPersistStateBetweenUsages()
```

Comment:
In writing my own with Qdrant, I kind of think using a Theory is cleaner, so we don't have to duplicate so much code.